### DISCUSSION TITLE
Make MoveAlongPath be reversible

### DISCUSSION THREAD

**Discussion by marcell-ziegler** [ManimCommunity/manim#3242] [Suggestions and Proposals]
I recently tried to make an animation that has a `Dot` moving along a graph one way, then the other. I noticed that there are some rate functions in the utilities which are reversed such as `there_and_back` but there are only two of these and for example there is no linear one. From what I can see the base class `Animation` has support for reversing any function with the `reverse_rate_func` parameter, but the inner workings of it are beyond me. This would in theory let you split the action into two animations in a `Sucession` for example, but `MoveAlongPath` does not seem to respect this parameter for some reason. I made a jury-rigged solution by subclassing:
```
class MoveAlongPathReverse(MoveAlongPath):
    @override
    def interpolate_mobject(self, alpha: float) -> None:
        point = self.path.point_from_proportion(self.rate_func(1 - alpha))
        self.mobject.move_to(point)
```
This works as intended for my use case but i am not confident enough to incorporate it into a proper pull request. Thus, I am leaving it here as a feature request if it is deemed worthy and someone on the dev team gets time to implement it properly.

**Comment by uwezi** [comment]
Is it really necessary to have a specialized function for everything? Here are two solutions to what you are asking for:

```
class thereAndBack1(Scene):
    def construct(self):
        sine = FunctionGraph(np.sin, x_range=[0,2*PI,0.01])
        self.add(sine)
        dot = Dot(radius=0.2,color=RED)
        self.add(dot)
        self.play(MoveAlongPath(dot,sine),rate_func=rate_functions.linear, run_time=2)
        self.wait()
        self.play(MoveAlongPath(dot,sine.reverse_direction()),rate_func=rate_functions.linear, run_time=2)
        self.wait()
```
```
class thereAndBack2(Scene):
    def construct(self):
        sine = FunctionGraph(np.sin, x_range=[0,2*PI,0.01])
        self.add(sine)
        dot = Dot(radius=0.2,color=RED)
        self.add(dot)
        self.play(MoveAlongPath(dot,sine),rate_func=rate_functions.linear, run_time=2)
        self.wait()
        self.play(MoveAlongPath(dot,sine),rate_func=lambda x: 1-rate_functions.linear(x), run_time=2)
        self.wait()
```

**Reply by marcell-ziegler** [reply]
Thank you! I didn't know you could reverse the points in a VMobject and I have no idea how I did not think of the lambda solution. Thanks for your suggestion.

**Reply by uwezi** [reply]
Come over to the Discord server, there you can get help and tips faster....
https://discord.com/channels/581738731934056449/1019649969596153968
